### PR TITLE
Add shield functionality

### PR DESCRIPTION
**Pull request by simonclausen**
Related to #7 - I needed it for myself so I just went ahead and wrote the state and templates. Depending on which direction this formula should go, this can also be implemented in an add-on formula.

But then again, as I mentioned in #7, I see shield as core functionality. And, no elasticsearch, no shield :)

P.S. because I did a late separate branch, this also includes the suggested change in #6.

### CODE REVIEW

**Comment by simonclausen** [comment]
Having worked a bit with these changes, I see that some adjustments need to be made before:
- Issues when doing first highstate, Shield configs are attempted applied before the plugin is installed
- Shield config changes does not require a restart of the elasticsearch service.

File: elasticsearch/shield.sls [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      - sls: elasticsearch.pkg
+
+elasticsearch_shield_roles_cfg:
+  file.managed:
+    - name: /etc/elasticsearch/shield/roles.yml
+    - source: salt://elasticsearch/files/shield/roles.yml
+    - template: jinja
+    - user: root 
+    - require:
+      - sls: elasticsearch.pkg
```
Comment (ryanwalder) [inline elasticsearch/shield.sls]: Could these not make use of `file.serialize`?

File: elasticsearch/service.sls [hunk 2]
```diff
@@ -1,12 +1,16 @@
 include:
   - elasticsearch.pkg
   - elasticsearch.config
+  - elasticsearch.shield
```
Comment (simonclausen) [inline elasticsearch/service.sls]: I agree

File: elasticsearch/shield.sls (hunk 1 above)
Comment (simonclausen) [inline elasticsearch/shield.sls]: Again, I agree that would probably be preferred, cheers

**Comment by simonclausen** [comment]
Quick follow up: I have been swamped with other things. I have made some of the mentioned changes in my local running formula, but I do not consider the changes of a sharable quality just yet. Hopefully I can come back with something in December.

**Comment by saintx** [comment]
Looking forward to reviewing it when it's ready, @simonclausen!

**Comment by simonclausen** [comment]
Ok, so almost half a year later I have time to take this up again. Also a lot of great stuff has been done to this repo. So I am closning this and will be starting over, of course taking the input here with me.

**Pull request closed without merging**